tests/test_logic.py

```diff
@@ -58,6 +58,19 @@ def test_complex_as_html(build_outcome: str) -> None:
     assert build_outcome
 
 
+@pytest.mark.sphinx(buildername="html", testroot="hook")
+def test_hook(build_outcome: str) -> None:
+    assert build_outcome
+
+
+@pytest.mark.sphinx(buildername="text", testroot="hook-fail")
+def test_hook_fail(app: SphinxTestApp) -> None:
+    app.build()
+    text = (Path(app.outdir) / "index.txt").read_text()
+    assert "Failed to hook argparse to get ArgumentParser" in app._warning.getvalue()
```

Comment:
I don't think you're supposed to read from a private field here. You also should extend the changelog.

Reply:
I couldn't really find any other way to check the errors, perhaps you know of another? Otherwise the test will just be an `assert text == ""`, but if that's fine by you then I can also just leave it at that.

Reply:
There must be a way, did you check what sphinx test suite does?

Reply:
You are entirely correct, it's even already used by this project.. :) I obviously looked in the wrong places.